### PR TITLE
[SCRIPTS] Introduce `setupvars` for Powershell

### PR DESCRIPTION
**Pull request by akashchi**
### Details:
* This PR introduces the `setupvars` script for Powershell. 
* As there are many places where the `setupvars.bat` version is used, this PR does not remove it. 
* The Windows GHA workflow is used as the first testing ground.

### Tickets:
 - *41421*

### CODE REVIEW

File: scripts/setupvars/setupvars.ps1 [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#   call "%INTEL_OPENVINO_DIR%\opencv\setupvars.bat"
+#   goto :opencv_done
+#)
+#
+#if exist "%INTEL_OPENVINO_DIR%\extras\opencv\setupvars.bat" (
+#   call "%INTEL_OPENVINO_DIR%\extras\opencv\setupvars.bat"
+#   goto :opencv_done
+#)
+#:opencv_done
+#
```
Comment (ilya-lavrenov) [inline scripts/setupvars/setupvars.ps1:22]: BTW, OpenCV can be dropped.

File: scripts/setupvars/setupvars.ps1 [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    Exit 1
+}
+
+if ($python_bitness -ne "64")
+{
+    Write-Host "Unsupported Python bitness. Please install one of Python $PYTHON_VERSION_MAJOR.$MIN_REQUIRED_PYTHON_VERSION_MINOR - $PYTHON_VERSION_MAJOR.$MAX_SUPPORTED_PYTHON_VERSION_MINOR (64-bit) from https://www.python.org/downloads/"
+    Exit 1
+}
+
+
```
Comment (akashchi) [inline scripts/setupvars/setupvars.ps1:123]:
```suggestion
```

File: scripts/setupvars/setupvars.ps1 [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+$Env:PYTHONPATH = "$Env:INTEL_OPENVINO_DIR/python;$Env:INTEL_OPENVINO_DIR/python/python3;$Env:PYTHONPATH"
+
+Write-Host "[setupvars.bat] OpenVINO environment initialized"
```
Comment (akashchi) [inline scripts/setupvars/setupvars.ps1:126]:
```suggestion
Write-Host "[setupvars] OpenVINO environment initialized"
```

File: scripts/setupvars/setupvars.ps1 [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+# Arguments parsing
+param (
+    [string]$python_version
+)
+
+$Env:INTEL_OPENVINO_DIR = Split-Path $MyInvocation.MyCommand.Path
+
+$Env:InferenceEngine_DIR = "$Env:INTEL_OPENVINO_DIR/runtime/cmake"
+$Env:ngraph_DIR = "$Env:INTEL_OPENVINO_DIR/runtime/cmake"
```
Comment (ilya-lavrenov) [inline scripts/setupvars/setupvars.ps1:12]: both `InferenceEngine_DIR` and `ngraph_DIR` are obsolete and will be removed in 2024.0 (current master)

Suggest dropping these variables.

**Review by ilya-lavrenov — APPROVED** [review]

File: cmake/developer_package/packaging/archive.cmake [hunk 5]
```diff
@@ -90,6 +90,7 @@ macro(ov_define_component_include_rules)
     # scripts
     unset(OV_CPACK_COMP_INSTALL_DEPENDENCIES_EXCLUDE_ALL)
     unset(OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL)
+    unset(OV_CPACK_COMP_SETUPVARS_PWSH_EXCLUDE_ALL)
```
[the remark this answers is not in the document]
Comment (akashchi) [reply cmake/developer_package/packaging/archive.cmake]: Yeah, that seems much better now, thanks.

File: cmake/developer_package/packaging/packaging.cmake [hunk 6]
```diff
@@ -152,6 +152,7 @@ macro(ov_define_component_names)
     # scripts
     set(OV_CPACK_COMP_INSTALL_DEPENDENCIES "install_dependencies")
     set(OV_CPACK_COMP_SETUPVARS "setupvars")
+    set(OV_CPACK_COMP_SETUPVARS_PWSH "setupvars_pwsh")
```
Comment (ilya-lavrenov) [inline cmake/developer_package/packaging/packaging.cmake:155]: let's drop this variable

File: scripts/CMakeLists.txt [hunk 7]
```diff
@@ -29,18 +29,18 @@ ov_cpack_add_component(${OV_CPACK_COMP_SETUPVARS} HIDDEN)
...
-        file(READ "${_setupvars_file_full_path}" _setupvars_content)
-        string(REPLACE "Release" ${CMAKE_BUILD_TYPE} _setupvars_content "${_setupvars_content}")
-        set(_setupvars_file_full_path "${OpenVINO_BINARY_DIR}/${_setupvars_file}")
-        message(STATUS "Writing patched content to ${_setupvars_file_full_path}")
-        file(WRITE "${_setupvars_file_full_path}" "${_setupvars_content}")
-    endif()
+
+    # setupvars.ps1
+    set(_setupvars_file_pwsh "setupvars/setupvars.ps1")
+    set(_setupvars_file_pwsh_full_path "${CMAKE_CURRENT_SOURCE_DIR}/${_setupvars_file_pwsh}")
```
Comment (ilya-lavrenov) [inline scripts/CMakeLists.txt:38]:
```suggestion
    list(APPEND _setupvars_files "${CMAKE_CURRENT_SOURCE_DIR}/${_setupvars_file_pwsh}")
```

File: scripts/CMakeLists.txt [hunk 8]
```diff
@@ -29,18 +29,18 @@ ov_cpack_add_component(${OV_CPACK_COMP_SETUPVARS} HIDDEN)
...
+
+    install(PROGRAMS "${_setupvars_file_pwsh_full_path}"
+            DESTINATION .
+            COMPONENT ${OV_CPACK_COMP_SETUPVARS}
+            ${OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL})
 endif()
 
 install(PROGRAMS "${_setupvars_file_full_path}"
```
Comment (ilya-lavrenov) [inline scripts/CMakeLists.txt:46]:
```suggestion
install(PROGRAMS ${_setupvars_files}
```

File: scripts/CMakeLists.txt [hunk 9]
```diff
@@ -29,18 +29,18 @@ ov_cpack_add_component(${OV_CPACK_COMP_SETUPVARS} HIDDEN)
 if(UNIX)
     set(_setupvars_file_full_path "${CMAKE_CURRENT_SOURCE_DIR}/setupvars/setupvars.sh")
 elseif(WIN32)
+    # setupvars.bat
     set(_setupvars_file "setupvars/setupvars.bat")
     set(_setupvars_file_full_path "${CMAKE_CURRENT_SOURCE_DIR}/${_setupvars_file}")
```
Comment (ilya-lavrenov) [inline scripts/CMakeLists.txt:34]:
```suggestion
    set(_setupvars_files "${CMAKE_CURRENT_SOURCE_DIR}/${_setupvars_file}")
```

File: scripts/CMakeLists.txt [hunk 10]
```diff
@@ -29,18 +29,18 @@ ov_cpack_add_component(${OV_CPACK_COMP_SETUPVARS} HIDDEN)
 if(UNIX)
     set(_setupvars_file_full_path "${CMAKE_CURRENT_SOURCE_DIR}/setupvars/setupvars.sh")
```
Comment (ilya-lavrenov) [inline scripts/CMakeLists.txt:30]:
```suggestion
    set(_setupvars_files "${CMAKE_CURRENT_SOURCE_DIR}/setupvars/setupvars.sh")
```

File: scripts/CMakeLists.txt [hunk 11]
```diff
@@ -29,18 +29,18 @@ ov_cpack_add_component(${OV_CPACK_COMP_SETUPVARS} HIDDEN)
...
-    endif()
+
+    # setupvars.ps1
+    set(_setupvars_file_pwsh "setupvars/setupvars.ps1")
+    set(_setupvars_file_pwsh_full_path "${CMAKE_CURRENT_SOURCE_DIR}/${_setupvars_file_pwsh}")
+
+    install(PROGRAMS "${_setupvars_file_pwsh_full_path}"
+            DESTINATION .
+            COMPONENT ${OV_CPACK_COMP_SETUPVARS}
+            ${OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL})
```
Comment (ilya-lavrenov) [inline scripts/CMakeLists.txt:43]: we can combine installation of several files into a single command and pass a list of files as input

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by raymondlo84** [comment]
Wow, is that going on the next release?

**Comment by ilya-lavrenov** [comment]
> Wow, is that going on the next release?

2024.0

**Pull request merged**